src/main.rs

```diff
@@ -20,69 +23,87 @@ struct Cli {
+    repo_head_sha: Option<String>,
+    #[arg(long, help = "Value to override branch of repository head.")]
+    repo_head_branch: Option<String>,
+    #[arg(long, help = "Value to override commit epoch of repository head.")]
+    repo_head_commit_epoch: Option<String>,
+    #[arg(
+        long,
+        value_delimiter = ',',
+        help = "Comma separated list of custom tag=value pairs."
+    )]
+    tags: Vec<String>,
+    #[arg(long, help = "Print files which will be uploaded to stdout.")]
+    print_files: bool,
+    #[arg(long, help = "Run metrics CLI without uploading to API.")]
+    dry_run: bool,
+}
+
+#[derive(Args, Clone, Debug)]
+struct TestArgs {
+    #[command(flatten)]
+    upload_args: UploadArgs,
+    #[arg(
+        required = false,
+        allow_hyphen_values = true,
+        trailing_var_arg = true,
+        help = "Test command to invoke."
+    )]
+    command: Vec<String>,
+}
+
 #[derive(Debug, Subcommand)]
 enum Commands {
-    #[clap(name = "upload")]
-    Upload {
-        #[arg(
-            long,
-            required = true,
-            value_delimiter = ',',
-            help = "Comma-separated list of glob paths to junit files."
-        )]
-        junit_paths: Vec<String>,
-        #[arg(long, help = "Organization url slug.")]
-        org_url_slug: String,
-        #[arg(
-            long,
-            required = true,
-            env = "TRUNK_API_TOKEN",
-            help = "Organization token. Defaults to TRUNK_API_TOKEN env var."
-        )]
-        token: String,
-        #[arg(long, help = "Path to repository root. Defaults to current directory.")]
-        repo_root: Option<String>,
-        #[arg(long, help = "Value to override URL of repository.")]
-        repo_url: Option<String>,
-        #[arg(long, help = "Value to override SHA of repository head.")]
-        repo_head_sha: Option<String>,
-        #[arg(long, help = "Value to override branch of repository head.")]
-        repo_head_branch: Option<String>,
-        #[arg(long, help = "Value to override commit epoch of repository head.")]
-        repo_head_commit_epoch: Option<String>,
-        #[arg(
-            long,
-            value_delimiter = ',',
-            help = "Comma separated list of custom tag=value pairs."
-        )]
-        tags: Vec<String>,
-        #[arg(long, help = "Print files which will be uploaded to stdout.")]
-        print_files: bool,
-        #[arg(long, help = "Run metrics CLI without uploading to API.")]
-        dry_run: bool,
-    },
+    Upload(UploadArgs),
+    Test(TestArgs),
 }
 
-const DEFAULT_API_ADDRESS: &str = "https://api.trunk.io/v1/metrics/createBundleUpload";
+const DEFAULT_ORIGIN: &str = "https://api.trunk.io";
 // Tokio-retry uses base ^ retry * factor formula.
 // This will give us 8ms, 64ms, 512ms, 4096ms, 32768ms
 const RETRY_BASE_MS: u64 = 8;
 const RETRY_FACTOR: u64 = 1;
-const RETRY_COUNT: usize = 5;
+const RETRY_COUNT: usize = 3;
```

Comment:
5 was causing very long delays when there were issues with the api.

Reply:
I've brought this up with Vlad before and he suggested keeping it at 5 because this gives the backend enough time to auto-resolve issues, expensive queries to finish, etc. Not sure if this is still a concern though cc @zaycev

Reply:
The delay is very long with 5 (10+ seconds). I think we need a lower number.

Reply:
Yeah, I'd mentioned that to Vlad as well. Will let him comment on this

Reply:
> 5 was causing very long delays when there were issues with the api.

That was the original point for adding this delay - wait enough for transient issue to resolve. What are the down sites of it?

Reply:
It feels broken when nothing happens for 20 seconds.